aait/backend/group-2B/Application/Features/Posts/Handlers/Commands/UpdatePostCommandHandler.cs

```diff
@@ -0,0 +1,42 @@
+using AAiT.Backend.G2B.SocialSync.Application.DTOs.Validators.PostDtoValidators;
+using AutoMapper;
+using MediatR;
+using SocialSync.Application.Contracts.Persistence;
+using SocialSync.Application.Features.Posts.Requests.Commands;
+using SocialSync.Application.Common.Responses;
+
+namespace SocialSync.Application.Features.Posts.Handlers.Commands;
+
+public class UpdatePostCommandHandler : PostsRequestHandler, IRequestHandler<UpdatePostCommand, BaseCommandResponse>
+{
+    public UpdatePostCommandHandler(IUnitOfWork unitOfWork, IMapper mapper) : base(unitOfWork, mapper)
+    {
+    }
+
+    public async Task<BaseCommandResponse> Handle(UpdatePostCommand request, CancellationToken cancellationToken)
+    {
+        var UpdateValidator = new UpdatePostDtoValidator(_userRepository, _postRepository);
+        var UpdateValidationResult = await UpdateValidator.ValidateAsync(request.UpdatePostDto);
+        var response = new BaseCommandResponse();
+        if (!UpdateValidationResult.IsValid)
+        {
+
+            response.Success = false;
```

Comment:
you have to make the result successsfull after you checked the changes you made have been saved to the database successfully